GhausAliKhan/Catalog-of-My-things
# Create class music and genre with required method and tests

**PR Description: Implement MusicAlbum and Genre Classes with Associated Functionality**

### Changes Made:

1. Created a new file, `MusicAlbum.rb`, to define the `MusicAlbum` class with all properties from the diagram properly initialized in the constructor method.

2. Implemented a separate file, `Genre.rb`, to define the `Genre` class. The class is associated with the `Item` class, and its properties are defined and set up in the constructor method.

3. Added an `add_item` method in the `Genre` class, which accepts an instance of the `Item` class as input. This method adds the input item to the collection of items, and sets `self` as a property of the item object using the appropriate setter from the item object.

4. Implemented a `can_be_archived?` method in the `MusicAlbum` class, overriding the method from the parent class. This method returns true if the parent's method returns true and if `on_spotify` equals true; otherwise, it returns false.

5. Added comprehensive unit tests for all the implemented methods to ensure robust functionality.

6. Provided options to:
   - List all music albums
   - List all genres (e.g., 'Comedy', 'Thriller')
   - Add a music album

7. Ensured data preservation by saving collections in `.json` files.

8. Created a `schema.sql` file with tables analogous to the structure of the classes:
   - `music_albums` table with all properties and associations from the parent `Item` class as table columns.
   - `genres` table.

### How to Test:

1. Run the unit tests to ensure the correctness of the implemented methods.
2. Use the provided options to list music albums and genres.
3. Add a new music album and verify data preservation in the `.json` files.
